pkg/controller/pxc/upgrade.go

```diff
@@ -78,8 +78,16 @@ func (r *ReconcilePerconaXtraDBCluster) updatePod(sfs api.StatefulApp, podSpec *
 
 	// pmm container
 	if cr.Spec.PMM != nil && cr.Spec.PMM.Enabled {
-		pmmC := sfs.PMMContainer(cr.Spec.PMM, cr.Spec.SecretsName, !cr.VersionLessThan120())
-		if !cr.VersionLessThan120() {
+		var versionLessThan120 bool
+		compare, err := cr.CompareVersion("1.2.0")
+		if err != nil {
+			return fmt.Errorf("compare version", err)
+		}
+		if compare == -1 {
+			versionLessThan120 = true
+		}
+		pmmC := sfs.PMMContainer(cr.Spec.PMM, cr.Spec.SecretsName, !versionLessThan120)
+		if !versionLessThan120 {
```

Comment:
1. `!versionLessThan120` looks too entangled. Better to express that as "greater or equal".
2. Why do we setup PMM container twice in different places? We pass "version" to `sfs.PMMContainer` so the result definitely depends on version. And then we do some additional PMM container' adjustments again based on version. All of these should be done in one place, in `sfs.PMMContainer()`.